### PR TITLE
Improve pre functions

### PR DESCRIPTION
**Pull request by zickgraf**
See commit messages for details.

Note that I have removed the "i.e." part from the description of `Lift` and `Colift` because otherwise the wording gets unclear.

### CODE REVIEW

**Comment by zickgraf** [comment]
CI is green, it just didn't get triggered because I started this as a draft PR:

https://circleci.com/gh/zickgraf/CAP_project/172

File: CAP/gap/CategoryMorphisms.gd [hunk 1]
```diff
@@ -1164,10 +1164,9 @@ DeclareOperation( "AddIsColiftableAlongEpimorphism",
                   [ IsCapCategory, IsList ] );
 
 #! @Description
-#! The arguments are two morphisms $\alpha: a \rightarrow c$, $\beta: b \rightarrow c$
-#! such that there is a lift $\alpha / \beta: a \rightarrow b$ of $\alpha$ along $\beta$, i.e.,
-#! a morphism such that $\beta \circ (\alpha / \beta) \sim_{a,c} \alpha$.
-#! The output is such a lift or $\mathtt{fail}$ if it doesn't exist.
+#! The arguments are two morphisms $\alpha: a \rightarrow c$, $\beta: b \rightarrow c$.
+#! The output is a lift $\alpha / \beta: a \rightarrow b$ of $\alpha$ along $\beta$
```
Comment (sebastianpos) [inline CAP/gap/CategoryMorphisms.gd]: Could you leave the explanation of what a lift is (the part in the original documentation after the` i.e.`). I find it helpful whenever I look up `Lift/Colift`.

Same for colifts.

Comment (zickgraf) [reply CAP/gap/CategoryMorphisms.gd]: Done

File: CAP/gap/MethodRecord.gi [hunk 2]
```diff
@@ -79,15 +215,49 @@ ProjectiveLift := rec(
...
+        
+    elif value = false then
+        
+        return [ false, "the two morphisms must have equal ranges" ];
+        
+    fi;
+    
+    return [ true ];
+  end,
+  return_type := "morphism",
```
Comment (sebasguts) [inline CAP/gap/MethodRecord.gi]: This seems wrong, as the documentation says this can return fail.

Comment (zickgraf) [reply CAP/gap/MethodRecord.gi]: Which part of the documentation are you referring to exactly? In my version, neither `ProjectiveLift` nor `AddProjectiveLift` mentions `fail`.

Comment (sebastianpos) [reply CAP/gap/MethodRecord.gi]: And if it did, it would simply be wrong.

Comment (sebasguts) [reply CAP/gap/MethodRecord.gi]: Sorry, I was probably looking at the docs for Lift/Colift.

**Pull request merged**